protocols/raft/src/main/java/io/atomix/protocols/raft/roles/AbstractAppender.java

```diff
@@ -447,7 +449,7 @@ protected InstallRequest buildInstallRequest(RaftMemberContext member, Snapshot
       try (SnapshotReader reader = snapshot.openReader()) {
         // Skip to the next batch of bytes according to the snapshot chunk size and current offset.
         reader.skip(member.getNextSnapshotOffset() * MAX_BATCH_SIZE);
-        final byte[] data = new byte[Math.min(MAX_BATCH_SIZE, reader.remaining())];
+        final ByteBuffer data = ByteBuffer.allocate(Math.min(MAX_BATCH_SIZE, reader.remaining()));
```

Comment:
How about pre allocate this buffer with direct memory?

Reply:
We can do that, I think the serialization code is already using `ByteBuffer#remaining` to know how much bytes to serialize, so it shouldn't be an issue to have a bigger buffer :+1:

Reply:
:+1: